src/julia_gc.c

```diff
@@ -728,20 +728,36 @@ void InitBags(UInt initial_size, Bag * stack_bottom, UInt stack_align)
 
     // Import GAPTypes module to have access to GapObj abstract type.
     // Needs to be done before setting any GC states
-    jl_eval_string("import GAPTypes");
-    if (jl_exception_occurred()) {
-        Panic("could not import GAPTypes module into Julia");
+    UInt            go_to_gaptypes_fallback = 0;
+    jl_module_t *   gaptypes_module;
+    jl_datatype_t * gapobj_type = (jl_datatype_t *)jl_get_global(
+        jl_main_module, jl_symbol("__GapObj_TYPE_VAR"));
+    if (!jl_exception_occurred()) {
+        go_to_gaptypes_fallback = 1;
     }
-    // Get GapObj abstract julia type
-    jl_module_t * gaptypes_module =
-        (jl_module_t *)jl_get_global(jl_main_module, jl_symbol("GAPTypes"));
-    if (jl_exception_occurred()) {
-        Panic("Could not read global GAPTypes variable from Julia");
+    else {
+        gaptypes_module = (jl_module_t *)jl_get_global(
+            jl_main_module, jl_symbol("__GapObj_MODULE_VAR"));
```

Comment:
So, how about we instead get the GAP.jl module, from the already existing global var `__JULIAGAPMODULE`. Then, if it exists, it will also have the GAPTypes module as a member, and from that we can get GapObj, no?

Reply:
Probably also wouldn't hurt to verify we are really getting a module, via `jl_is_module`